### PR TITLE
Update OpenSSL pod and remove exclude arch setting

### PR DESCRIPTION
**Pull request by priteshrnandgaonkar**
## Summary

This PR updates the OpenSSL pod and thus updates the corresponding Folly and RSocket pods. The new OpenSSL pod has a support of arm64 artifact for iPhomeSimulators.This PR also updates the replaces the libevent pod which didn't work with arm64 arch for iPhoneSimulator.

## Changelog

- Update OpenSSL pod
- Update Folly dep
- Update RSocket dep
- Replace Libevent pod.

## Test Plan

CI

### CODE REVIEW

File: iOS/Podspecs/libevent.podspec [hunk 1]
```diff
@@ -0,0 +1,565 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Because libevent is also relied on by Flipper and *with* OpenSSL bindings,
```
Comment (alloy) [inline iOS/Podspecs/libevent.podspec:6]: So I take it this isn’t correct then? Seeing as I don’t see any changes to enable the libevent OpenSSL bindings, or did I overlook them?

**Review by alloy — APPROVED** [review]
Nice 👌 

Have you verified if RN and Flipper will indeed share the dependency? Either way, in the future it might be a good idea to start distributing the libevent podspec via the central repository.

File: iOS/Podspecs/libevent.podspec (hunk 1 above)
Comment (priteshrnandgaonkar) [reply iOS/Podspecs/libevent.podspec:6]: Yeah its nothing was required to be changed. I will remove that comment from the podspec file. Also you can now refer this pod from cocoapod direcctly. I have published it in the cocoapod repository.

Comment (alloy) [reply iOS/Podspecs/libevent.podspec]: > I have published it in the cocoapod repository.

Ah perfect! :pray:

**Comment by Fl0p** [comment]
+1

**Comment by hsavit1** [comment]
Having trouble with this, discussion here https://github.com/facebook/flipper/issues/1758#issuecomment-777691398

**Pull request closed without merging**